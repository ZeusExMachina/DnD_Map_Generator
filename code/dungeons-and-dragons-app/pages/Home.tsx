import React, { useState } from 'react';
import {View} from "react-native";
import LargeButton from "../components/LargeButton";
import {Typography} from "@material-ui/core";


const Home = () => {
    return (
        <View>
            <div className={"backgroundImage"}>
                <div className={"homeBox"}>
                    <Typography variant={"h2"} className={"homeTitle"}>
                        What are you ..?
                    </Typography>
                    <div className={"homeButtons"}>
                        <LargeButton buttonString={"Player"} buttonRoute={"/playeroptions"} leftAndRightPadding={110}/>
                        <LargeButton buttonString={"Dungeon Master"} buttonRoute={"/dmoptions"} leftAndRightPadding={25}/>
                    </div>
                </div>
            </div>
        </View>

    );
}

export default Home

