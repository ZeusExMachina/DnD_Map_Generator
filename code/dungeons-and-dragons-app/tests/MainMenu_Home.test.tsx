import React from 'react';
import renderer from 'react-test-renderer'

import Home from "../pages/Home";

/*Home Page Rendering*/
describe('<MainMenu_Home />', () => {

    test('Home page renders correctly', () => {
        const renderedComponent = renderer.create(<Home />).toJSON()
        expect(renderedComponent).toMatchSnapshot();
    });
    
});