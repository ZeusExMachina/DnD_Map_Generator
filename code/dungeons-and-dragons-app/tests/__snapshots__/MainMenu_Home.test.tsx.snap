// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainMenu_Home /> Home page renders correctly 1`] = `
<View>
  <div
    className="backgroundImage"
  >
    <div
      className="homeBox"
    >
      <h2
        className="MuiTypography-root homeTitle MuiTypography-h2"
      >
        What are you ..?
      </h2>
      <div
        className="homeButtons"
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-outlined largeButton MuiButton-outlinedPrimary MuiButton-outlinedSizeLarge MuiButton-sizeLarge"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            <h4
              className="MuiTypography-root makeStyles-buttonText-2 MuiTypography-h4"
            >
              Player
            </h4>
          </span>
        </button>
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-outlined largeButton MuiButton-outlinedPrimary MuiButton-outlinedSizeLarge MuiButton-sizeLarge"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            <h4
              className="MuiTypography-root makeStyles-buttonText-2 MuiTypography-h4"
            >
              Dungeon Master
            </h4>
          </span>
        </button>
      </div>
    </div>
  </div>
</View>
`;
